detectron2/modeling/proposal_generator/rpn.py

```diff
@@ -431,24 +431,25 @@ def forward(
             for x in pred_anchor_deltas
         ]
 
-        if self.training:
+        if self.training and not torch.jit.is_scripting():
             gt_labels, gt_boxes = self.label_and_sample_anchors(anchors, gt_instances)
             losses = self.losses(
                 anchors, pred_objectness_logits, gt_labels, pred_anchor_deltas, gt_boxes
             )
-            losses = {k: v * self.loss_weight.get(k, 1.0) for k, v in losses.items()}
+            for k, v in losses.items():
+                losses[k] = v * self.loss_weight.get(k, 1.0)
```

Comment:
please add a note linked to a pytorch feature request about dict comprehension, and boolean in dictionary.

Reply:
done